packages/react-data-grid/src/masks/InteractionMasks.js

```diff
@@ -452,23 +582,38 @@ class InteractionMasks extends React.Component {
             columns={columns}
           />
         )}
-        {!isEditorEnabled && this.isGridSelected() && (
-          <SelectionMask
-            selectedPosition={selectedPosition}
+        {selectedRangeIsSingleCell(this.state.selectedRange) ? (
```

Comment:
I think it could be nice to split this chunk of jsx into simple sub components for readability and to simplify the conditional logic.

```javascript
{selectedRangeIsSingleCell(this.state.selectedRange) ?
<SingleCellSelectView {...relavantProps/> : <CellRangeSelectView {...otherRelevantProps}/>
```